File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        self.ans = []
        
        def InorderTravesal(parent):
            
            if not parent:
                return 
            
            left = InorderTravesal(parent.left)
            self.ans.append(parent.val)
            right = InorderTravesal(parent.right)
            
        InorderTravesal(root)
        return (self.ans)

File: 0094-binary-tree-inorder-traversal/0094-binary-tree-inorder-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        
        vector<int> ans;
        
        function<void(TreeNode*)> dfs = [&](TreeNode* node){
          if(!node) return;
            
          dfs(node->left);
          ans.push_back(node->val);
          dfs(node->right);
        };
        dfs(root);
        
        return ans;
    }
};